### ISSUE TITLE
Detecting exponential growth computeds 

### ISSUE THREAD

**Issue by brianmhunt** [knockout/tko#55]
A fairly common problem in my experience is the exponential growth of computed re-computations, particularly in large projects with lots of complex interdependencies.

Sometimes `deferred` computeds resolve the problem, but sometimes they are not ideal or possible.

I'd like to diagnose when it is a performance issue, and further to be able to isolate it.

My first thought is to overload `ko.computed` with a rate-sampling on recomputation so e.g.

```javascript
// globally
const INTERVAL = 60 * 1000
setInterval(checkTicks, INTERVAL)
let ticks = 0

function checkTicks () {
   if (ticks > SOME_RATE) {
     console.info("A computed is very active")
   }
   ticks = 0
}

// overload the computeds' readFunction so
state.readFunction = () => {
   ticks ++
   theOriginalReadFunction()
}
```

This is the most basic part, simply telling the user that there's a large number of recomputations.

It may be possible to use the computeds `_version` for this, and similarly have an extender (but part of the issue is globally identifying what computed is the problem).

@knockout/core thoughts?

**Comment by mbest** [comment]
Is this something that Chrome's (or other browser's) profiler can help with?

**Comment by brianmhunt** [comment]
Not really, from my experience.  Since observables are anonymous, one just sees the chains of notifications without any indication of origin.

Having tinkered, it looks like overloading `computed.fn.updateVersion` with some sort of frequency checker may be a good way to go.

**Comment by daedalus28** [comment]
On a related note, an API like mobx's [`trace`](https://mobx.js.org/best/trace.html) tells you _why_ the recompute happened which is very helpful.
